### PR TITLE
Extend test coverage to excercise most parameter choices.

### PR DESCRIPTION
**Pull request by mlopatka**
Added some more test coverage. Outstanding TODOs indicated in comments.

### CODE REVIEW

File: tests/test_generate_dpdata.py [hunk 1]
```diff
@@ -20,8 +20,52 @@ def test_defaults():
...
+    dp_data = test_ron_gauss_instance.generate_dpdata(test_data, 1, test_labels)
+    assert normality_test_helper(dp_data)
+
+
+def test_simple_supervised_data():
+    test_data = data_gen_helper(1000, 3)
+    test_labels = np.random.choice([0, 1], size=100, p=[1./2, 1./2])
+    test_ron_gauss_instance = ron_gauss.RONGauss(algorithm="supervised")
+    # TODO: Specify a reasonable value for maxY that excercises the code in supervised condition.
+    dp_data = test_ron_gauss_instance.generate_dpdata(X=test_data,
```
Comment (mlopatka) [inline tests/test_generate_dpdata.py:66]: @tchanyaswad This test fails, I'm not sure what a reasonable set of parameters and simulated data would be to properly exercise the execution that is specific to the "supervised" code when `algorithm="supervised"` is specified. Can you refine this test case?

Comment (tchanyaswad) [reply tests/test_generate_dpdata.py:66]: Let me try running this and I can see what's done wrong here. One quick hunch is that `maxY` might have to be float, not an array to work, but I will confirm again.

Comment (mlopatka) [reply tests/test_generate_dpdata.py:66]: @tchanyaswad any chance you have a passing test for `ron_gauss.RONGauss(algorithm="supervised")`?

In my next PR I'm planning to break out test coverage into functional and statistical tests as per Issue #4

Comment (tchanyaswad) [reply tests/test_generate_dpdata.py:66]: Found a passing test for "supervised". Please change `test_labels = np.random.choice([0, 1], size=100, p=[1./2, 1./2]) ` to `test_labels = np.random.choice([0, 1], size=1000, p=[1./2, 1./2]) `.

The sample size of `test_labels` and `test_data` has to be the same, i.e. `len(test_labels) == len(test_data)`

Comment (mlopatka) [reply tests/test_generate_dpdata.py:66]: OMG, I can't believe I missed that. I need to sleep more :)
Thanks for the spot check. PR incoming!

**Pull request merged**